app/controllers/api/v1/timeframe_versions_controller.rb

```diff
@@ -0,0 +1,45 @@
+class Api::V1::TimeframeVersionsController < Api::BaseController
+  skip_before_action :verify_authenticity_token
+  before_action :set_page, only: :index
+
+  def index
+    if from_time.nil? || to_time.nil?
+      render plain: 'timeframe parameters must be iso8601 formatted',
+             status: :bad_request
+    else
+      render_rubygems(version_timeframe_query)
+    end
+  end
+
+  private
+
+  def from_time
+    @parse_from_time ||= Time.iso8601(params.require(:from))
```

Comment:
why do we need `||=` here?

Reply:
It's just some memoization so that the parse only happens once, even though `from_time` is called twice.